File: src/lib/functions/notesTools.ts

import { FunctionDeclaration, SchemaType } from "@google/generative-ai";

export const updateUserNotesTool: FunctionDeclaration = {
  name: "updateUserNotes",
  description: "Updates the user's general notes with new information. Use this to remember miscellaneous facts or context provided by the user.",
  parameters: {
    type: SchemaType.OBJECT,
    properties: {
      notes: {
        type: SchemaType.STRING,
        description: "The new content to be saved in the user's notes. This will overwrite existing notes.",
      },
    },
    required: ["notes"],
  },
};

File: mise-asi/registry/schemas/notes_tools.py
"""
Notes tool schemas
Maps to: src/lib/functions/notesTools.ts
"""

update_user_notes_tool = {
    "name": "updateUserNotes",
    "description": "Updates the user's personal notes. This is an overwrite operation - the new notes replace any existing notes.",
    "input_schema": {
        "type": "object",
        "properties": {
            "notes": {"type": "string", "description": "The new notes content"}
        },
        "required": ["notes"]
    }
}
